### DISCUSSION TITLE
Will there be support for Xamarin.UITest or will there be some framework that replaces it?

### DISCUSSION THREAD

**Discussion by rory-bloomerang** [dotnet/maui#3702] [Q&A]
I can't seem to find any answers on any of the FAQ pages or blog posts regarding this - can we expect the Xamarin.UITest framework to work with MAUI or will there be some replacement framework that works similarly?

I would love to keep using the UITest framework with MAUI - but I don't see any information on whether or not that will be possible. What is the plan for automated UI testing in MAUI?

**Comment by davidbuckleyni** [comment]
Agreed they have Shaun away from this but i guess its a given, but things like app centre for diagnostics and testing they did not say a bird about.

**Comment by JohnHDev** [comment]
Xamarin UITest works with the native controls that Xamarin and Maui adds to the native views, so theoretically it should still work. As per a reply from David: https://devblogs.microsoft.com/dotnet/announcing-net-maui-preview-10/#comment-14201
